plugins/discovery/pod.go

```diff
@@ -27,28 +27,39 @@ func newPodHandler(kubeClient kubernetes.Interface, discoverer discovery.Discove
 
 	inf.AddEventHandler(cache.ResourceEventHandlerFuncs{
 		AddFunc: func(obj interface{}) {
-			pod := obj.(*v1.Pod)
-			podUpdated(pod, discoverer)
+			updatePodIfValid(obj, discoverer)
```

Comment:
Couldn't we do this by guarding against all bad casting conditions like this?
```
pod, ok := obj.(*v1.Pod)
if ok { podUpdated( pod, discoverer ) }
```
Then we guard against this and all other bad downcasting scenarios at once?

Reply:
Done